README


This archive contains the distribution Dist-Zilla-Plugin-ModuleBuildTiny-Fallback,
version 0.014:

  Build a Build.PL that uses Module::Build::Tiny, falling back to Module::Build as needed

This software is copyright (c) 2014 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.025.



Changes
Revision history for Dist-Zilla-Plugin-ModuleBuildTiny-Fallback

0.014     2014-11-14 03:19:43Z
          - and another (sigh)

0.013     2014-11-14 01:24:36Z
          - avoid more test failures with older [ModuleBuildTiny]

0.012     2014-11-12 05:34:24Z
          - avoid test failure with older [ModuleBuildTiny]

0.011     2014-11-11 06:45:37Z
          - now checks if all configure prereqs are installed, not just
            Module::Build::Tiny
          - warn the user if something else added content to Build.PL,
            which might make it not a good idea to use this plugin

0.010     2014-11-02 05:47:11Z
          - keep Build.PL in the file list between phases to allow other
            plugins to add to its content

0.009     2014-10-02 04:55:27Z
          - more fixes as in 0.008 and 0.007

0.008     2014-09-28 01:59:03Z
          - fix test failure caused by unstated assumption that
            Dist::Zilla::Role::TestRunner is a very recent version

0.007     2014-09-14 01:36:01Z
          - fix breakage caused by unstated assumption that
            Dist::Zilla::Role::TestRunner is a very recent version

0.006     2014-09-06 01:18:50Z
          - now dumping config data for [MetaConfig], including the configs
            used by [ModuleBuild] and [ModuleBuildTiny] (composed by this
            plugin)
          - ensure 'use strict; use warnings;' remains near the top of
            Build.PL, as some plugins use it as a matching point for adding
            new content

0.005     2014-06-08 06:12:39Z
          - lowered default minimum Module::Build version from 0.3601 to 0.28,
            to handle more legacy systems; documented the configuration
            options better

0.004     2014-06-06 05:37:01Z
          - add 'mb_version', 'mbt_version' options, for specifying the
            minimum required version of Module::Build and Module::Build::Tiny

0.003     2014-05-29 21:47:58Z
          - skip [ModuleBuild]'s configure_requires declaration, because if
            Module::Build is running, the prereqs were being ignored anyway

0.002     2014-05-23 07:47:57Z
          - minor doc fixes
          - handle some potential future changes to [ModuleBuild],
            [ModuleBuildTiny]

0.001     2014-05-22 07:31:17Z
          - Initial release.


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-ModuleBuildTiny-Fallback was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.016
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.31.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Build a Build.PL that uses Module::Build::Tiny, falling back to Module::Build as needed",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-ModuleBuildTiny-Fallback",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::ModuleBuildTiny::Fallback",
  "PREREQ_PM" => {
    "Dist::Zilla::Plugin::ModuleBuild" => 0,
    "Dist::Zilla::Plugin::ModuleBuildTiny" => 0,
    "Dist::Zilla::Role::BeforeBuild" => 0,
    "Dist::Zilla::Role::BuildPL" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "MooseX::Types" => 0,
    "MooseX::Types::Moose" => 0,
    "Scalar::Util" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0,
    "Test::Requires" => 0,
    "if" => 0
  },
  "VERSION" => "0.014",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Dist::Zilla::Plugin::ModuleBuild" => 0,
  "Dist::Zilla::Plugin::ModuleBuildTiny" => 0,
  "Dist::Zilla::Role::BeforeBuild" => 0,
  "Dist::Zilla::Role::BuildPL" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::PrereqSource" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "List::Util" => 0,
  "Module::Build::Tiny" => "0.039",
  "Moose" => 0,
  "MooseX::Types" => 0,
  "MooseX::Types::Moose" => 0,
  "Path::Tiny" => 0,
  "Scalar::Util" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "Test::Requires" => 0,
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL
# This Build.PL for Dist-Zilla-Plugin-ModuleBuildTiny-Fallback was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.014
use strict;
use warnings;

my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-ModuleBuildTiny-Fallback was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.008.
    use strict;
    use warnings;

    use 5.006;
    # use Module::Build::Tiny 0.039;
    Module::Build::Tiny::Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.025.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "Module::Build::Tiny" => "0.039",
        "perl" => "5.006"
      },
      "dist_abstract" => "Build a Build.PL that uses Module::Build::Tiny, falling back to Module::Build as needed",
      "dist_author" => [
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-ModuleBuildTiny-Fallback",
      "dist_version" => "0.014",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::ModuleBuildTiny::Fallback",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Dist::Zilla::Plugin::ModuleBuild" => 0,
        "Dist::Zilla::Plugin::ModuleBuildTiny" => 0,
        "Dist::Zilla::Role::BeforeBuild" => 0,
        "Dist::Zilla::Role::BuildPL" => 0,
        "Dist::Zilla::Role::FileGatherer" => 0,
        "Dist::Zilla::Role::PrereqSource" => 0,
        "List::Util" => 0,
        "Moose" => 0,
        "MooseX::Types" => 0,
        "MooseX::Types::Moose" => 0,
        "Scalar::Util" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.006",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "Path::Tiny" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => 0,
        "Test::Requires" => 0,
        "if" => 0,
        "perl" => "5.006"
      }
    );


    my %fallback_build_requires = (
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "Path::Tiny" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => 0,
      "Test::Requires" => 0,
      "if" => 0,
      "perl" => "5.006"
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
